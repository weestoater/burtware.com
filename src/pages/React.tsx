import React from "react";
import ReactHooksPost from "../components/react/ReactHooksPost";
import RandomUsersPost from "../components/react/RandomUserPost";

export default function ReactPage() {
  return (
    <div className="row">
      <RandomUsersPost />
      <ReactHooksPost />
      <div className="col-lg-3 col-md-6 mb-4">
        <div className="card">
          <div className="card-header">
            <i className="bi bi-filetype-jsx" />
            <span className="pl-4">my experiences</span>
          </div>
          <div className="card-body">
            <p>
              There are as many ways to do Single Page Applications (SPAs) as
              there are developers with opinions. Within my work we have many
              instances of AngularJS, Angular (4,5,6,7,9,11,12) and ReactJS.
            </p>
            <p>
              My experience with ReactJS began as pretty limited, but with a
              strong following wind, a few cups of coffee and a Pluralsight
              subscription â€“ I resolved that in no small measure.</p>
            <p>
              Before moving recently to pure UX team, I worked on a growing React platform which included multiple chart and grid offerings.
            </p>
            <p>
              I am now trying to get my head around the worlds of <code>typescript</code>; micro-services and AWS with React.
            </p>
          </div>
        </div>
      </div>
    </div>
  );
}
