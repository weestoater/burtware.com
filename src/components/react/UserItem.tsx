import React from "react";

function UserItem(props: any) {

  const user = props;

  return (
    <>
      <div className="user col-sm-12 mb-3">
        <img src={user.item.picture.large} alt="" className="picture" />
        <p className="name">
          <span className="badge bg-info text-black me-2">{user.item.nat}</span>{" "}
          {user.item.name.title} {user.item.name.first} {user.item.name.last}
        </p>
        <address className="address">
          {user.item.location.street.number} {user.item.location.street.name}{" "}
          <br />
          {user.item.location.city}
          <br />
          {user.item.location.state}, {user.item.location.country}
          <br />
          {user.item.location.postcode}
        </address>
        <p className="contact">
          <i className="bi bi-phone" aria-hidden="true" />{" "}
          <a href={`tel:${user.item.cell}`}>{user.item.cell}</a>
          <br />
          <i className="bi bi-telephone" aria-hidden="true" />{" "}
          <a href={`tel:${user.item.phone}`}>{user.item.phone}</a>
          <br />
          <i className="bi bi-envelope" aria-hidden="true" />{" "}
          <a href={`mailto:${user.item.email}`}>{user.item.email}</a>
        </p>
      </div>
    </>
  );
}

export default UserItem;
