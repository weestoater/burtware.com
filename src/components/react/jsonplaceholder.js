import React, { useState, useEffect } from 'react';

export const JsonPlaceholder = () => {

    // const fetchData = fetch('https://geodb-cities-graphql.p.rapidapi.com', {
    //     method: 'POST',
    //     headers: {'Content-Type': 'application/json'},
    //     body: JSON.stringify({
    //         query:`{
    //             countries {
    //                 name
    //                 capital
    //                 currency
    //             }
    //         }`
    //     })
    // })
    // .then(res => res.json())
    // .then(res => console.log(res.data));

    // fetchData();

    return (
        <div className="card">
            <div className="card-header">FETCH DATA</div>
            <div className="card-body">
                body    
            </div> 
        </div>
    );
}