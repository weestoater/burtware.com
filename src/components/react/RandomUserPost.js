import React, {useState, useEffect} from 'react';
import getApi from '../../adapters/commonApi';
import Spinner from '../global/Spinner';
import UserItem from './UserItem';

export default function RandomUsersPost() {

    const [users, setUsers] = useState({});
    const apiUrl = "https://randomuser.me/api/?results=3";

    useEffect(() => {
        getApi(apiUrl).then(setUsers);
      }, []);

    return (
        <div className="col-lg-4 col-md-6 mb-4">
            <div className="card">
                <div className="card-header">
                    <i className="bi bi-filetype-jsx" />
                    <span className="pl-4">Random Users API</span>
                </div>
                <div className="card-body users">
                {users.results ? (
                    users.results.map((user, key) => <UserItem key={key} item={user} /> )
                ) : ( 
                <Spinner /> 
                )}
                </div>
            </div>
        </div>
    );
}
