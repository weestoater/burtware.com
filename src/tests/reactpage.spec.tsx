import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import ReactPage from "../pages/React";

describe('Testing React page', () => {

    it('can render ReactPage', () => {
        const {container} = render(<ReactPage />)
        expect(container).toMatchSnapshot();
    });
});