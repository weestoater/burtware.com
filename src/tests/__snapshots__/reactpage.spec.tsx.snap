// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing React page can render ReactPage 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-sm-12"
    >
      <h1
        class="visually-hidden"
      >
        React stuff
      </h1>
    </div>
    <div
      class="col-lg-4 col-md-6 mb-4"
    >
      <div
        class="card"
      >
        <div
          class="card-header"
        >
          <i
            class="bi bi-filetype-jsx"
          />
          <span
            class="pl-4"
          >
            Random Users API
          </span>
        </div>
        <div
          class="card-body users"
        >
          <div
            class="row"
          >
            <div
              class="d-flex justify-content-center"
            >
              <div
                class="spinner-border"
                role="status"
              />
            </div>
            <br />
            <span
              class="text-center"
            >
              Loading...
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-lg-4 col-md-6 mb-4"
    >
      <div
        class="card"
      >
        <div
          class="card-header"
        >
          <i
            class="bi bi-filetype-jsx"
          />
          <span
            class="pl-4"
          >
            hooks
          </span>
        </div>
        <div
          class="card-body"
        >
          <p>
            Since moving teams and re-learning Angular I’ve had the chance to work recently on a rather nice ReactJS project. It gave me my first real exposure to full stack development and I have to say, whilst terrified, I felt well chuffed when it was all done – with help of a couple of amazing colleagues.
          </p>
          <p>
            The really sweet part was using React Hooks for the first time. Rather than wrestling with state and setState it was just a simple useEffect and useState which was a revelation. So nice.
          </p>
          <p>
            Never felt entirely comfortable mixing mark-up and functions, that was a bit of a hurdle to overcome. Then the whole state and setState really grinds my gears too.
          </p>
          <div
            class="code-sample"
          >
            <p>
              fetch('/api/getNachos')
              <br />
                .then(response =&gt; response.json())
              <br />
                .then(data =&gt; this.setState({data}));
            </p>
          </div>
          <p>
            Instead it’s a nice simple case of creating a fetch.js service file for carrying out the api calls. Then a really clean and simple few lines to set up the object for using on the page.
          </p>
          <div
            class="code-sample"
          >
            <p>
              const [data, setData] = useState(null);
              <br />
              useEffect( () =&gt; {
              <br />
                  getData().then(setData);
              <br />
              }, []);
            </p>
          </div>
          <p>
            I personally find this a lot cleaner and nicer to work with. A plus side is also the test-ability of it, which matters to me as it’s a key part of my employers ethos.
          </p>
        </div>
      </div>
    </div>
    <div
      class="col-lg-4 col-md-6 mb-4"
    >
      <div
        class="card"
      >
        <div
          class="card-header"
        >
          <i
            class="bi bi-filetype-jsx"
          />
          <span
            class="pl-4"
          >
            my experiences
          </span>
        </div>
        <div
          class="card-body"
        >
          <p>
            There are as many ways to do Single Page Applications (SPAs) as there are developers with opinions. Within my work we have many instances of AngularJS, Angular (4,5,6,7,9,11,12) and ReactJS.
          </p>
          <p>
            My experience with ReactJS began as pretty limited, but with a strong following wind, a few cups of coffee and a Pluralsight subscription – I resolved that in no small measure.
          </p>
          <p>
            Before moving recently to pure UX team, I worked on a growing React platform which included multiple chart and grid offerings.
          </p>
          <p>
            I am now trying to get my head around the worlds of 
            <code>
              typescript
            </code>
            ; micro-services and AWS with React.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
